lib/schema/date.js

```diff
@@ -232,9 +232,12 @@ SchemaDate.prototype.cast = function(value) {
 
   if (value instanceof Number || typeof value === 'number') {
     date = new Date(value);
+  } else if (typeof value === 'string' && !isNaN(Number(value)) && (Number(value) >= 275761 || Number(value) < 0)) {
```

Comment:
I'm not 100% sold on this idea but I like it in general. Nice compromise between using the 'Date' constructor where possible and falling back to the pre #5880 behavior when it makes sense. However, instead of `Number(value) < 0`, let's do `Number(value) < MIN_YEAR` because `new Date('-2017')` is perfectly valid in JS and I don't want to break the date behavior.

Reply:
I agree, it could cause confusion to someone who's trying to do something with the first 5 minutes of 1970.

Reply:
done